File: src/2586.count-the-number-of-vowel-strings-in-range.2654/solution.ts
function vowelStrings (words: string[], left: number, right: number): number {
    let result = 0;
    const set = new Set(['a', 'e', 'i', 'o', 'u', ]);
    for (let i = left; i <= right; i++) {
        const word = words[i];
        if (set.has(word[0]) && set.has(word[word.length - 1])) {
            result++;
        }
    }
    return result;
}

File: src/2586.count-the-number-of-vowel-strings-in-range.2654/solution.java
class Solution {
    public int vowelStrings(String[] words, int left, int right) {
        int result = 0;
        char[] chars = new char[] { 'a', 'e', 'i', 'o', 'u' };

        while (left <= right) {
            boolean l = false;
            boolean r = false;
            for (char c : chars) {
                if (words[left].charAt(0) == c) {
                    l = true;
                }
                if (words[left].charAt(words[left].length() - 1) == c) {
                    r = true;
                }
            }

            if (l && r) {
                result++;
            }

            left++;
        }
        return result;
    }
}